walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,80 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
+/** Finds possible meeting times for a request for all mandatory participants */
 public final class FindMeetingQuery {
+  /**
+   * @param events List of all known events with times and participants
+   * @param request Meeting to be planned
+   * @return List of all possible meeting times
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<TimeRange> busyTimes = getBusyTimes(events, request);
+    Collections.sort(busyTimes, TimeRange.ORDER_BY_START);
+    busyTimes = mergeIntervals(busyTimes);
+    return getAvailableTimes(busyTimes, request.getDuration());
+  }
+
+  // gets busy times for all request attendees
+  private ArrayList<TimeRange> getBusyTimes(Collection<Event> events, MeetingRequest request) {
```

Comment:
It's probably good to return a type as generic as possible (just like your other private methods returning List) unless you actually need ArrayList-specific functionality

Reply:
done!